crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -4563,6 +4561,112 @@ impl<'db> TypeInferenceBuilder<'db> {
         }
     }
 
+    /// Rich comparison in Python are the operators `==`, `!=`, `<`, `<=`, `>`, and `>=`. Their
+    /// behaviour can be edited for classes by implementing corresponding dunder methods.
+    /// This function performs rich comparison between two instances and returns the resulting type.
+    /// see `<https://docs.python.org/3/reference/datamodel.html#object.__lt__>`
+    fn infer_rich_comparison(
```

Comment:
have these functions changed at all, or are they just moved to become methods in this PR?

Reply:
I didn't change them (other than that I renamed them). I thought it's not worth another PR but I don't want this change to be part of my `__bool__` PR. But I should have called it out!

Reply:
No worries! I'm fine with them being moved, just wanted to check that there wasn't any other change being made here that I needed to review 😁